reactor-core/src/main/java/reactor/util/context/Context.java

```diff
@@ -132,9 +167,42 @@ static Context of(Object key1, Object value1,
 			Object key3, Object value3,
 			Object key4, Object value4,
 			Object key5, Object value5) {
+		checkDuplicateKeys(key1, key2, key3, key4, key5);
 		return new Context5(key1, value1, key2, value2, key3, value3, key4, value4, key5, value5);
 	}
 
+	/**
+	 * Create a {@link Context} out of a {@link Map}. Prefer this method if you're somehow
+	 * incapable of checking keys are all distinct in other {@link #of(Object, Object, Object, Object, Object, Object, Object, Object)}
+	 * implementations.
+	 *
+	 * @implNote this method compacts smaller maps into a relevant fields-based implementation
+	 * when number of key-value pairs is under 6.
+	 */
+	static Context of(Map<?, ?> keyValuePairs) {
```

Comment:
the variable name is a bit misleading because there are no "pairs", map is a map :) Maybe just "values"?

Reply:
map is a container of key-value pairs, isn't it? but yeah, I should simply say `when size of the map is less than 6`.

Reply:
(my bad thought this referred to the javadoc comment... I'll also change the parameter name)